internal/resource/common/id/id_schema.go

```diff
@@ -20,14 +20,15 @@ func ToSchema(s *schema.Schema) {
 	s.Attributes["id"] = schemaId
 }
 
-func ToSchemaCustomId(s *schema.Schema, idName string, characterLimit bool, description string) {
+func ToSchemaCustomId(s *schema.Schema, idName string, required, characterLimit bool, description string) {
 	customId := schema.StringAttribute{}
 	customId.Description = description
 	customId.PlanModifiers = []planmodifier.String{
 		stringplanmodifier.UseStateForUnknown(),
 		stringplanmodifier.RequiresReplace(),
 	}
-	customId.Required = true
+	customId.Required = required
+	customId.Optional = !required
```

Comment:
Should this be computed as well? Does PF generate an ID when you don't specify one?

Reply:
yes, it does generate one

Reply:
updated